### ISSUE TITLE
withApiAuthRequired does not accept authorization header

### ISSUE THREAD

**Issue by softshipper** [auth0/nextjs-auth0#388]
Hi all 

I have protected the GraphQL server in the API with the function `withApiAuthRequired` as the following:
```
const server = new ApolloServer({
    schema,
    context: () => new Context()
});
const handler = server.createHandler({path: "/api/graphql"});

export const config = {
    api: {
        bodyParser: false,
    },
};

export default withApiAuthRequired((req: NextApiRequest, res: NextApiResponse<void>) => {
    return handler(req, res);
});
```
When I make a request with GraphQL client to the server with the following HTTP header:
```
 {headers: {"Authorization": `Bearer ${session.accessToken}`}}
```
I have got the response `{ '1928170296': { error: { graphQLErrors: [], networkError: 'Error: Unauthorized' } } }`. 

The full code of the client implementation:
```
const session = await getSession(req, res);
const ssrCache = ssrExchange({isClient: false});
const client = initUrqlClient({
    url: addr.gql,
    exchanges: [dedupExchange, cacheExchange, ssrCache, fetchExchange],
    fetchOptions: () => {
        return session.accessToken === undefined ? {} : {headers: {"Authorization": `Bearer ${session.accessToken}`}}
    }
}, false);
```
The GraphQL client can not be authenticated to the server.

Best regards,
Softshipper

**Comment by adamjmcgrath** [comment]
Hi @softshipper - `withApiAuthRequired` is for protecting API routes with a cookie session. This SDK does not protect routes with a Bearer token. 

The idea is that you should use this SDK to login to auth0, which establishes an httpOnly session cookie, then the client can make requests to the api routes with the session cookie (not with the bearer authorization header)

The API route can then decrypt the cookie to get an access token and use that to access external APIs that are protected with a JWT bearer token.

See https://github.com/auth0/nextjs-auth0/blob/main/ARCHITECTURE.md#protect-an-api-route for more info

**Comment by softshipper** [comment]
@adamjmcgrath First of all, thanks for your answer. 
The solution would be to add the session cookie to the GraphQL client right? 
How can I get the session cookie from `const session = await getSession(req, res);`?

**Comment by adamjmcgrath** [comment]
Hi @softshipper - if you make an http request from the browser (with the graphql client) to a url on the same origin (eg `/api/graphql` will be on the same origin) then the cookie will already be on the request.

So as long as you are logged in, `withApiAuthRequired` should allow the request to pass to the route handler.

**Comment by softshipper** [comment]
Hi @adamjmcgrath
I would like to make the request to API inside the `getServerSideProps` hook as follows :
```
export const getServerSideProps: GetServerSideProps = withPageAuthRequired({
    async getServerSideProps({req, res}: GetServerSidePropsContext<ParsedUrlQuery>) {

        const client = initUrqlClient({
            url: addr.gql,
            exchanges: [dedupExchange, cacheExchange, ssrCache, fetchExchange],
        }, false);


    }
});
```
It is server to server request not browser to the server. Is it possible to do server to server request to the protected `API(withApiAuthRequired)`?
Why do I request to API inside `getServerSideProps` is to get the benefit of SSR prerendering.  

Thanks

**Comment by softshipper** [comment]
@adamjmcgrath I realized, what I want to do does not make sense. 
Thanks

**Comment by keriwarr** [comment]
Hi, I just want to add for posterity that I think what's being described here does make sense - we want to server side render an authenticated page using data that is protected by an authenticated api endpoint.

It's the same thing as what's being displayed in [the ultimate guide](https://auth0.com/blog/ultimate-guide-nextjs-authentication-auth0/)  in this image: ![serverless-server-api-call](https://images.ctfassets.net/23aumh6u8s0i/3FfAL4aBTfb5VotjE2mAbo/02270036429ea2db708b74ebe4b8e2b8/serverless-server-api-call)

The solution is described here by adamjmcgrath: https://github.com/auth0/nextjs-auth0/issues/288#issuecomment-777680609

tl;dr: whatever logic you're executing in the api endpoint that you're trying to call, you should _directly_ execute the same logic inside of `getServerSideProps`
